src/util/cluster.js

```diff
@@ -1,30 +1,21 @@
 const cluster = require('cluster');
 const gracefulShutdown = require('http-graceful-shutdown');
-const util = require('util');
 const logger = require('../logger');
+const { logProcessInfo } = require('./utils');
 
 const numWorkers = parseInt(process.env.NUM_PROCS || '1', 10);
 
-function processInfo() {
-  return {
-    pid: process.pid,
-    ppid: process.ppid,
-    mem: process.memoryUsage(),
-    cpu: process.cpuUsage(),
-    cmd: `${process.argv0} ${process.argv.join(' ')}`,
-  };
-}
-
 function finalFunction() {
-  logger.info(`worker (pid: ${process.pid}) was gracefully shutdown`);
+  logger.error(`Worker (pid: ${process.pid}) was gracefully shutdown`);
+  logProcessInfo();
 }
 
 // This function works only in master.
 // It sends SIGTERM to all the workers
 function shutdownWorkers() {
   Object.values(cluster.workers).forEach((worker) => {
     process.kill(worker.process.pid);
-    logger.info(`Sent kill signal to worker ${worker.id} (pid: ${worker.process.pid})`);
+    logger.error(`Sent kill signal to worker ${worker.id} (pid: ${worker.process.pid})`);
```

Comment:
How do we make a difference b/w graceful shutdown vs forced shutdown if this log is marked as default `error` ?

Reply:
Graceful shutdown is only for the HTTP server but not the process itself. After a process receives a terminate signal, we have a grace period for the server to complete its pending actions.
I've made it an error to be able to log this message always, irrespective of the log level.